legal-api/src/legal_api/services/filings/validations/alteration.py

```diff
@@ -120,3 +122,31 @@ def type_change_validation(filing):
                     'path': legal_type_path})
         return msg
     return []
+
+
+def rules_change_validation(filing):
+    """Validate rules change."""
+    msg = []
+    rules_file_key: Final = get_str(filing, '/filing/alteration/rulesFileKey')
+    rules_file_name: Final = get_str(filing, '/filing/alteration/rulesFileName')
+
+    if rules_file_key or rules_file_name:
+        if not (rules_file_key and rules_file_name):
+            msg.append({'error': babel('Both rulesFileKey and rulesFileName should be provided'),
+                        'path': '/filing/alteration/rulesFileKey and /filing/alteration/rulesFileName'})
+            return msg
+    return []
+
+
+def memorandum_change_validation(filing):
+    """Validate memorandum change."""
+    msg = []
+    memorandum_file_key: Final = get_str(filing, '/filing/alteration/memorandumFileKey')
+    memorandum_file_name: Final = get_str(filing, '/filing/alteration/memorandumFileName')
+
+    if memorandum_file_key or memorandum_file_name:
+        if not (memorandum_file_key and memorandum_file_name):
+            msg.append({'error': babel('Both memorandumFileKey and memorandumFileName should be provided'),
+                        'path': '/filing/alteration/memorandumFileKey and /filing/alteration/memorandumFileName'})
```

Comment:
can we provide the path for the specific memorandum property missing?  same for rules above